### DISCUSSION TITLE
Using Xpra to solve HiDPI issue for legacy X11 apps on wayland

### DISCUSSION THREAD

**Discussion by saivert** [Xpra-org/xpra#4419] [Ideas]
Seems like both KDE and Gnome now supports running Xwayland without scaling everything. This means that games and X11 apps that is DPI-aware will scale propertly without blurry graphics.
But this causes legacy X11 apps to display very small. I have been experimenting with using Xpra to solve this since the way it integrated the windows actually worked somewhat well. Xpra comes with `run_scaled` utility so this is easy to test.

But I'm having an issue where Xpra just quits if I click anywhere inside a X11 window. I guess it has trouble running on Wayland.

Other software I have tried that kind of does the same:
- wayland-proxy-virtwl
- nested rootful Xwayland using xwayland-run

Would this be a way to keep legacy support?

**Comment by totaam** [comment]
> But I'm having an issue where Xpra just quits if I click anywhere inside a X11 window.

Never heard of anything like it.
Are you sure you are not using [dowstream packages](https://github.com/Xpra-org/xpra/wiki/Distribution-Packages)?

> I guess it has trouble running on Wayland.

Not as far as I know.
Only some more minor things are more difficult or just impossible on Wayland.

> Would this be a way to keep legacy support?

If by legacy, you mean X11 scaling, yes.
